src/utils/components/DiskModal/utils/helpers.ts

```diff
@@ -131,7 +131,10 @@ export const getDataVolumeFromState = ({
   createOwnerReference?: boolean;
 }): V1beta1DataVolume => {
   const dataVolume = getEmptyVMDataVolumeResource(vm, createOwnerReference);
-  const dvName = resultVolume?.dataVolume?.name || `${vm?.metadata?.name}-${diskState.diskName}`;
+  const dvName =
+    resultVolume?.dataVolume?.name ||
+    resultVolume?.persistentVolumeClaim?.claimName ||
+    `${vm?.metadata?.name}-${diskState.diskName}`;
```

Comment:
I think we shouldn't need `?` in `vm?.metadata?.name` here. If yes then something is going wrong in here, because I think that..  if we think that it might be `undefined` (because we add "?" in there), then the name wouldn't look nice/correct. Or would it be correct anyway? Ignore this comment if yes. But if not, it's better to let the UI break here so the issue could be seen and fixed asap.

Reply:
I cannot fully agree, only partially. This is what you want most of the time. But from my experience (and from experience of some other skilled devs), this is too defensive development style that can lead to situations when the UI doesn't crash even when it better should - in cases when something is not done well in that place, and then absence of the UI crash leads to not revealing a bug early enough, and causing more problems that can be hard to find later.